modules/data.land/R/IC_SOILGRID_Utilities.R

```diff
@@ -0,0 +1,348 @@
+  if (!requireNamespace("MASS", quietly = TRUE)) {
+    PEcAn.logger::logger.severe("MASS package required for SoilGrids ensemble generation")
+  }
+  if (verbose) {
+    PEcAn.logger::logger.info("Preprocessing soil carbon data following PEcAn standards")
+  }
+
+  # Only process sites with complete mean data for both depths
+  complete_sites <- !is.na(soil_data$`Total_soilC_0-30cm`) & 
+                   soil_data$`Total_soilC_0-30cm` > 0 &
+                   !is.na(soil_data$`Total_soilC_0-200cm`) & 
+                   soil_data$`Total_soilC_0-200cm` > 0
+
+  if (!any(complete_sites)) {
+    PEcAn.logger::logger.severe("No sites with complete data for both depth intervals found")
+  }
+  
+  processed <- soil_data[complete_sites, ]
+  
+  if (verbose) {
+    removed_count <- nrow(soil_data) - nrow(processed)
+    PEcAn.logger::logger.info(sprintf("Removed %d site(s) with incomplete data. Processing %d sites", 
+                                    removed_count, nrow(processed)))
+  }
+  
+  # Calculate CV distributions
+  depths <- list(
+    "30cm" = list(mean_col = "Total_soilC_0-30cm", std_col = "Std_soilC_0-30cm"),
+    "200cm" = list(mean_col = "Total_soilC_0-200cm", std_col = "Std_soilC_0-200cm")
+  )
+  
+  cv_dist <- lapply(depths, function(depth_info) {
+    valid_cv <- processed[[depth_info$mean_col]] > 0 & 
+               !is.na(processed[[depth_info$std_col]]) & 
+               processed[[depth_info$std_col]] > 0
+    
+    if (sum(valid_cv) < 5) {
+      return(list(type = "none"))
+    }
+    
+    cv_values <- processed[[depth_info$std_col]][valid_cv] / processed[[depth_info$mean_col]][valid_cv]
+    cv_bounds <- stats::quantile(cv_values, probs = c(0.05, 0.95), na.rm = TRUE)
+    cv_filtered <- cv_values[cv_values >= cv_bounds[1] & cv_values <= cv_bounds[2]]
+    
+    if (length(cv_filtered) < 5) {
+      return(list(type = "none"))
+    }
+    
+    gamma_fit <- try(MASS::fitdistr(cv_filtered, "gamma"), silent = TRUE)
+    if (!inherits(gamma_fit, "try-error")) {
+      list(
+        type = "gamma",
+        shape = gamma_fit$estimate["shape"],
+        rate = gamma_fit$estimate["rate"],
+        bounds = as.vector(cv_bounds)
+      )
+    } else {
+      list(
+        type = "empirical",
+        values = cv_filtered,
+        bounds = as.vector(cv_bounds)
+      )
+    }
+  })
+  
+  return(list(
+    data = processed,
+    cv_distribution_30cm = cv_dist[["30cm"]],
+    cv_distribution_200cm = cv_dist[["200cm"]]
+  ))
+}
+
+#' Generate soil carbon ensemble members for specific depth
+#'
+#' @param processed_data Output from preprocess_soilgrids_data()
+#' @param site_id Target site ID
+#' @param size Number of ensemble members to generate
+#' @param depth_layer Depth layer ("0-30cm" or "0-200cm")
+#' @param verbose Logical, print detailed progress information
+#' @param seed Optional random seed for reproducibility
+#' 
+#' @return Vector of soil carbon values with proper uncertainty handling
```

Comment:
Please say explicitly what handling is done that makes this "proper" (it might be best in the body of the function description instead of in the return value, though)